articles/cosmos-db/create-mongodb-golang.md

```diff
@@ -0,0 +1,279 @@
+1.  In the left menu, click **New**, click **Databases**, and then click **Azure
+Cosmos DB**.
+
+![](https://cdn-images-1.medium.com/max/800/1*e-QMvThW-2QZ3hEH3VvUQQ.png)
+
+3. In the **New account** blade, specify the desired configuration for the Azure
+Cosmos DB account.
+
+With Azure Cosmos DB, you can choose one of four programming models: Gremlin
+(graph), MongoDB, SQL (DocumentDB), and Table (key-value).
+
+In this quick start we’ll be programming against the MongoDB API so you’ll
+choose **MongoDB** as you fill out the form. But if you have graph data for a
+social media mes New app, document data from a catalog app, or key/value (table)
+data, realize that Azure Cosmos DB can provide a highly available,
+globally-distributed database service platform for all your mission-critical
+applications.
+
+Fill out the New account blade using the information in the screenshot as a
+guide . You will choose unique values as you set up your account so your values
+will not match the screenshot exactly
+
+![](https://cdn-images-1.medium.com/max/800/1*XdFlpeNKLe7o3FOtCZ17AQ.png)
+<span class="figcaption_hack">Create Azure Cosmos DB Account</span>
+
+4. Click **Create** to create the account.
+
+5. On the toolbar, click **Notifications** to monitor the deployment process.
+
+![](https://cdn-images-1.medium.com/max/800/1*WTGVkPwkSTxjUzKm2u09Eg.png)
+<span class="figcaption_hack">Check Progress of Azure Cosmos DB Deployment</span>
+
+6. Click on **golang-couch** resources.
+
+7. Get connection string information which will be required by client
+applications.
+
+![](https://cdn-images-1.medium.com/max/800/1*XoslPMsWsv5vqK2vhYAJPw.png)
+<span class="figcaption_hack">Connection information of MongoDB</span>
+
+### Setting up your application
+
+It’s time to make our hands dirty. Open your favorite editor (Gogland, VS Code
+or Atom). For this article, I will use Gogland editor.
+
+1.  Create folder CosmosDBAcces folder inside GOROOT\src folder
+1.  Run below command to get mgo package
+
+    go get gopkg.in/mgo.v2
+
+### Connect to an Azure Cosmos DB account
+
+[mgo](http://labix.org/mgo) (pronounced as *mango*) is a
+[MongoDB](http://www.mongodb.org/) driver for the [Go
+language](http://golang.org/) that implements a rich and well tested selection
+of features under a very simple API following standard Go idioms.
+
+Azure Cosmos DB uses latest version of MongoDB v3.2.0. with SSL enabled.
+
+To check MongoDb version go to Azure portal and get MongoDB Shell Tab and copy
+mongo command
+
+![](https://cdn-images-1.medium.com/max/800/1*voTl_TbS7Wuld2Q4CBLFXQ.png)
+<span class="figcaption_hack">Azure Cosmos MongoDB Shell</span>
+
+Open command prompt and run command. Once the connection is established, run
+below command to check database version.
+
+    db.version()
+
+![](https://cdn-images-1.medium.com/max/800/1*OJwDs_lG45z_K3rzw2_RuQ.png)
+
+Officially, mgo does not support MongoDB 3.2 version. There is solution to
```

Comment:
mgo supports TLS/SSL using the DialInfo or tls.Dial function.  More details at https://github.com/go-mgo/mgo/issues/84